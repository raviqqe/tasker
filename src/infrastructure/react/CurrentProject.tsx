import { defaultImport } from "default-import";
import defaultStyled from "styled-components";
import { IProject } from "../../domain/project.js";
import { Loader } from "./Loader.js";
import { white } from "./style/colors.js";

const styled = defaultImport(defaultStyled);

const Container = styled.div`
  display: flex;
  align-items: center;
  color: ${white};
  font-size: 1.6rem;
  cursor: pointer;
  word-break: break-word;
  height: 4rem;
`;

export interface IProps {
  currentProject: IProject | null;
  showProjects: () => void;
}

export const CurrentProject = ({
  currentProject,
  showProjects,
}: IProps): JSX.Element => (
  <Container onClick={showProjects}>
    {currentProject ? currentProject.name : <Loader />}
  </Container>
);
