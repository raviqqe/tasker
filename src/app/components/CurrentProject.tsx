import { styled } from "@linaria/react";
import { useStore } from "@nanostores/react";
import { projectPresenter } from "../../main/project-presenter.js";
import { Loader } from "./Loader.js";
import { white } from "../style.js";

const Container = styled.div`
  display: flex;
  align-items: center;
  color: ${white};
  font-size: 1.6rem;
  cursor: pointer;
  word-break: break-word;
  height: 4rem;
`;

export interface Props {
  onShowProjects: () => void;
}

export const CurrentProject = ({ onShowProjects }: Props): JSX.Element => {
  const project = useStore(projectPresenter.currentProject);

  return (
    <Container onClick={onShowProjects}>
      {project?.name ?? <Loader />}
    </Container>
  );
};
