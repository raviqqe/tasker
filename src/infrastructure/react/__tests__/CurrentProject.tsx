import { create } from "react-test-renderer";
import React from "react";
import { CurrentProject } from "../CurrentProject";

it("renders", () => {
  expect(
    create(
      <CurrentProject
        currentProject={{ archived: false, id: "", name: "" }}
        showProjects={() => {}}
      />
    ).toJSON()
  ).toMatchSnapshot();
});
