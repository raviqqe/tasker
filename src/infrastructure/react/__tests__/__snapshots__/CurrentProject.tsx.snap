// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-bdVaJa hqwmVs"
>
  <div
    className="sc-bwzfXH cMgkiV"
    onClick={[Function]}
  >
    
  </div>
</div>
`;
